client/components/timezone-dropdown/index.jsx

```diff
@@ -3,50 +3,79 @@
  * External Dependencies
  */
 import React from 'react';
-import noop from 'lodash/noop';
 
 /**
  * Internal dependencies
  */
+import wpcom from 'lib/wp';
 import Dropdown from 'components/select-dropdown';
-import i18n from 'lib/mixins/i18n';
 
-export default React.createClass( {
-	displayName: 'TimezoneDropdown',
+/**
+ * Module variables
+ */
+const { Component, PropTypes } = React;
+const noop = () => {};
+const undocumented = wpcom.undocumented();
+
+class TimezoneDropdown extends Component {
+	constructor() {
+		super();
 
-	propTypes: {
```

Comment:
https://facebook.github.io/react/docs/reusable-components.html#es6-classes

> ```es6
export class Counter extends React.Component {
  constructor(props) {
    super(props);
    this.state = {count: props.initialCount};
  }
  tick() {
    this.setState({count: this.state.count + 1});
  }
  render() {
    return (
      <div onClick={this.tick.bind(this)}>
        Clicks: {this.state.count}
      </div>
    );
  }
}
Counter.propTypes = { initialCount: React.PropTypes.number };
Counter.defaultProps = { initialCount: 0 };
```

Also I've seen this pattern in some code written by @aduth. am I right?